examples/taproot-psbt.rs

```diff
@@ -0,0 +1,556 @@
+                    vout: 0,
+                },
+                script_sig: Script::new(),
+                sequence: 0xFFFFFFFD, // enable locktime and opt-in RBF
+                witness: Witness::default(),
+            }],
+            output: vec![],
+        };
+        let mut next_psbt = Psbt::from_unsigned_tx(next_tx)?;
+        let mut origins = BTreeMap::new();
+        origins.insert(
+            beneficiary_key,
+            (
+                vec![leaf_hash],
+                (
+                    self.beneficiary_xpub.fingerprint(),
+                    derivation_path,
+                ),
+            ),
+        );
+        let ty = PsbtSighashType::from_str("SIGHASH_ALL")?;
+        let mut tap_scripts = BTreeMap::new();
+        tap_scripts.insert(taproot_spend_info.control_block(&(script.clone(), LeafVersion::TapScript)).unwrap(), (script, LeafVersion::TapScript));
+
+        let input = Input {
+            witness_utxo: {
+                let script_pubkey = script_pubkey;
+                let amount = Amount::from_sat(value);
+    
+                Some(TxOut {
+                    value: amount.to_sat(),
+                    script_pubkey,
+                })
+            },
+            tap_key_origins: origins,
+            tap_merkle_root: taproot_spend_info.merkle_root(),
+            sighash_type: Some(ty),
+            tap_internal_key: Some(internal_keypair.public_key()),
+            tap_scripts,
+            ..Default::default()
+        };
+
+        next_psbt.inputs = vec![input];
+        self.next_psbt = Some(next_psbt.clone());
+
+        self.next += 1;
+        Ok((tx, next_psbt))
+    }
+
+    fn refresh_tx(&self, locktime_delta: u32) -> Result<(Transaction, Psbt), Box<dyn std::error::Error>> {
+        todo!()
+    }
+}
+
+/// A wallet that allows spending from an inheritance locked to a P2TR UTXO via a script path
+/// after some expiry using CLTV.
+struct BeneficiaryWallet {
+    master_xpriv: ExtendedPrivKey,
+    secp: secp256k1::Secp256k1<secp256k1::All>
+}
+
+impl BeneficiaryWallet {
+    fn new(master_xpriv: ExtendedPrivKey) -> Result<Self, Box<dyn std::error::Error>> {
+        Ok(Self { master_xpriv, secp: Secp256k1::new() })
+    }
+
+    fn master_xpub(&self) -> ExtendedPubKey {
+        ExtendedPubKey::from_priv(&self.secp, &self.master_xpriv)
+    }
+
+    fn spend_inheritance(&self, mut psbt: Psbt, locktime: u32, to_address: Address, absolute_fees_in_sats: u64) -> Result<Transaction, Box<dyn std::error::Error>> {
+        let input_value = psbt.inputs[0].witness_utxo.as_ref().unwrap().value;
+        let input_script_pubkey = psbt.inputs[0].witness_utxo.as_ref().unwrap().script_pubkey.clone();
+        psbt.unsigned_tx.lock_time = locktime;
+        psbt.unsigned_tx.output = vec![TxOut {
+            script_pubkey: to_address.script_pubkey(),
+            value: input_value - absolute_fees_in_sats,
+        }];
+        psbt.outputs = vec![Output { ..Default::default() }];
```

Comment:
```suggestion
        psbt.outputs = vec![Output::default()];
```

Reply:
Of course :man_facepalming: :sweat_smile: